### PR TITLE
fix constant E in mpc and mpfr and also add tests

### PR DESCRIPTION
**Pull request by abhinavagarwal07**
fix https://github.com/symengine/symengine/issues/701
- [x] fix eval_mpc.cpp and eval_mpfr.cpp
- [x] add tests

### CODE REVIEW

File: symengine/eval_mpc.cpp [hunk 1]
```diff
@@ -231,10 +231,14 @@ class EvalMPCVisitor : public BaseVisitor<EvalMPCVisitor> {
             mpfr_clear(t);
         } else if (x.__eq__(*E)) {
             mpfr_t t;
-            mpfr_init2(t, mpc_get_prec(result_));
-            mpfr_const_euler(t, rnd_);
+            mpfr_t _one;
+            mpfr_init2 (_one, GMP_RNDD);
+            mpfr_init2(t, mpc_get_prec(result_));            
+            mpfr_set_d (_one, 1.0, GMP_RNDD);
```
Comment (isuruf) [inline symengine/eval_mpc.cpp]: Use `mpfr_set_ui` here as `1.0` might not equal to 1 at `prec`

File: symengine/eval_mpc.cpp [hunk 2]
```diff
@@ -231,10 +231,14 @@ class EvalMPCVisitor : public BaseVisitor<EvalMPCVisitor> {
             mpfr_clear(t);
         } else if (x.__eq__(*E)) {
             mpfr_t t;
-            mpfr_init2(t, mpc_get_prec(result_));
-            mpfr_const_euler(t, rnd_);
+            mpfr_t _one;
```
Comment (isuruf) [inline symengine/eval_mpc.cpp]: You only need one `mpfr_t` variable. Remove `_one`

File: symengine/eval_mpfr.cpp [hunk 3]
```diff
@@ -212,7 +212,11 @@ class EvalMPFRVisitor : public BaseVisitor<EvalMPFRVisitor> {
         if (x.__eq__(*pi)) {
             mpfr_const_pi(result_, rnd_);
         } else if (x.__eq__(*E)) {
-            mpfr_const_euler(result_, rnd_);
+            mpfr_t _one;
+            mpfr_init2 (_one, GMP_RNDD);
+            mpfr_set_d (_one, 1.0, GMP_RNDD);
```
Comment (isuruf) [inline symengine/eval_mpfr.cpp]: Use `mpfr_set_ui`

File: symengine/eval_mpfr.cpp [hunk 4]
```diff
@@ -212,7 +212,11 @@ class EvalMPFRVisitor : public BaseVisitor<EvalMPFRVisitor> {
         if (x.__eq__(*pi)) {
             mpfr_const_pi(result_, rnd_);
         } else if (x.__eq__(*E)) {
-            mpfr_const_euler(result_, rnd_);
+            mpfr_t _one;
+            mpfr_init2 (_one, GMP_RNDD);
```
Comment (isuruf) [inline symengine/eval_mpfr.cpp]: You need to give the precision here not `GMP_RNDD`

File: symengine/eval_mpc.cpp (hunk 2 above)
Comment (abhinavagarwal07) [inline symengine/eval_mpc.cpp]: then what to use instead of `_one` in `mpfr_exp(t, _one, rnd_)`

File: symengine/eval_mpc.cpp (hunk 1 above)
Comment (abhinavagarwal07) [inline symengine/eval_mpc.cpp]: Ok.

**Comment by abhinavagarwal07** [comment]
I will look at it later.

**Pull request closed without merging**